# 
# Submitted-By: Robert Schwebel, 2006-01-14
# Committed-By: Robert Schwebel
#
# Error: 
#
# obj_kallsyms.c: In function 'obj32_kallsyms':
# obj_kallsyms.c:204: error: invalid lvalue in assignment
# obj_kallsyms.c:279: error: invalid lvalue in assignment
# make[3]: *** [obj_kallsyms.o] Error 1
#
# Description: 
#
# gcc-4 doesn't understand the syntax being used here. For the second
# patch, see http://dev.gentoo.org/~vanquirius/gcc4-porting-guide.html
#
# State:
# 
# unknown
#

diff -urN modutils-2.4.27/obj/obj_kallsyms.c modutils-2.4.27-ptx/obj/obj_kallsyms.c
--- modutils-2.4.27/obj/obj_kallsyms.c	2002-03-01 01:39:06.000000000 +0100
+++ modutils-2.4.27-ptx/obj/obj_kallsyms.c	2006-01-14 14:41:32.000000000 +0100
@@ -200,8 +200,8 @@
 
     /* Initial contents, header + one entry per input section.  No strings. */
     osec->header.sh_size = sizeof(*a_hdr) + loaded*sizeof(*a_sec);
-    a_hdr = (struct kallsyms_header *) osec->contents =
-    	xmalloc(osec->header.sh_size);
+    osec->contents = xmalloc(osec->header.sh_size);
+    a_hdr = (struct kallsyms_header *)osec->contents;
     memset(osec->contents, 0, osec->header.sh_size);
     a_hdr->size = sizeof(*a_hdr);
     a_hdr->sections = loaded;
@@ -275,8 +275,8 @@
 	a_hdr->symbol_off +
 	a_hdr->symbols*a_hdr->symbol_size +
 	strings_size - strings_left;
-    a_hdr = (struct kallsyms_header *) osec->contents =
-	xrealloc(a_hdr, a_hdr->total_size);
+    osec->contents = xrealloc(a_hdr, a_hdr->total_size);
+    a_hdr = (struct kallsyms_header*)osec->contents;
     p = (char *)a_hdr + a_hdr->symbol_off;
     memcpy(p, symbols, a_hdr->symbols*a_hdr->symbol_size);
     free(symbols);
diff -urN modutils-2.4.27-orig/insmod/insmod.c modutils-2.4.27/insmod/insmod.c
--- modutils-2.4.27-orig/insmod/insmod.c	2003-10-27 03:34:46.000000000 +0100
+++ modutils-2.4.27/insmod/insmod.c	2006-01-14 14:48:51.000000000 +0100
@@ -275,7 +275,7 @@
 		if (strncmp((char *)s->name, "GPLONLY_", 8) == 0) {
 			gplonly_seen = 1;
 			if (gpl)
-				((char *)s->name) += 8;
+				s->name = ((char *)s->name) + 8;
 			else
 				continue;
 		}
diff -urN modutils-2.4.27/depmod/depmod.c modutils-2.4.27-orig/depmod/depmod.c
--- modutils-2.4.27/depmod/depmod.c	2006-01-14 14:55:26.000000000 +0100
+++ modutils-2.4.27-orig/depmod/depmod.c	2003-03-23 03:34:28.000000000 +0100
@@ -1133,7 +1133,7 @@
 
 		for (ksym = ksyms; so_far < nksyms; ++so_far, ksym++) {
 			if (strncmp((char *)ksym->name, "GPLONLY_", 8) == 0)
-				ksym->name = ((char *)ksym->name) + 8;
+				((char *)ksym->name) += 8;
 			assert(n_syms < MAX_MAP_SYM);
 			symtab[n_syms++] = addsym((char *)ksym->name, mod, SYM_DEFINED, 0);
 		}
diff -urN modutils-2.4.27/genksyms/genksyms.c modutils-2.4.27-orig/genksyms/genksyms.c
--- modutils-2.4.27/genksyms/genksyms.c	2006-01-14 14:54:16.000000000 +0100
+++ modutils-2.4.27-orig/genksyms/genksyms.c	2002-03-01 01:39:06.000000000 +0100
@@ -45,7 +45,7 @@
 int flag_debug, flag_dump_defs, flag_warnings;
 int checksum_version = 1, kernel_version = version(2,0,0);
 
-int errors;
+static int errors;
 static int nsyms;
 
 static struct symbol *expansion_trail;
